bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -1712,7 +1747,7 @@ protected void saveState() {
 
             if (currentURI != null) {
                 if (isPlayingStream(currentURI) || isPlayingRadioStartedByAmazonEcho(currentURI)
-                        || isPlayingRadio(currentURI)) {
+                        || isPlayingRadio(currentURI) || isPlayingRadioApp(currentURI)) {
```

Comment:
Can these four conditions be extracted to private method for being reused by line 2691-2692? It seems that the logic is similar, so would be candidates for some future developer updating one of them, but missing the other.

Reply:
Good idea. Done.